pythia/modules/fusions.py

```diff
@@ -0,0 +1,134 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+"""
+The fusions module contains various Fusions techniques some based on BLOCK:
+Bilinear Superdiagonal Fusion for VQA and VRD. For e.g. LinearSum, ConcatMLP
+taken from https://github.com/Cadene/block.bootstrap.pytorch#fusions.
+
+For implementing your own fusion technique, you need to follow these steps:
+
+.. code::
+    from pythia.common.registry import registry
+    from pythia.modules.fusions import BaseFusion
+    from torch import nn
+
+    @regitery.register_fusion("custom")
+    class CustomFusion(nn.Module):
+        def __init__(self, params=None):
+            super().__init__("Custom")
+
+Then in your model's config you can specify ``fusion`` attribute as follows:
+
+.. code::
+
+   model_attributes:
+      some_model:
+          fusion:
+             - type: block
+             - params: {}
+"""
+import collections
+
+import torch
+import torch.nn as nn
+from block import fusions
+
+from pythia.common.registry import registry
+
+
+@registry.register_fusion("block")
```

Comment:
Instead of doing this, you can just do 

`registry.register_fusion("block")(fusion.Block)` not the no `@` here.